mainbot.py

```diff
@@ -1,33 +1,110 @@
+import slack
+import asyncio
+import logging
 from slacker import Slacker
 from typing import List
 from datetime import datetime, timedelta
 
 
-def sort_messages(messages: List[dict], topk: int = 20) -> List[dict]:
+slacker: Slacker
+CRAWL_INTERVAL: int = 60 * 15  # in seconds
+
+
+async def crawl_messages() -> None:
+    def write_to_db_mock(anything):
+        print(anything, end="\n\n")
+
+    while True:
+        # get data
+        ch_info = await slacker.get_channels_list()
+        for ch_id, ch_name in ch_info:
+            print(f"Channel: {ch_name}")
+
+            day_ago = datetime.now() - timedelta(days=1)
+            messages = await slacker.get_channel_messages(ch_id, day_ago)
+            write_to_db_mock(messages)
+
+        # wait for next time
+        await asyncio.sleep(CRAWL_INTERVAL)
+
+
+def sort_messages(
+    messages: List[dict], topk: int = 20, criterion: str = "replies"
+) -> List[dict]:
     """
     Sort messages by unique and mysterious algorithm and return most important
 
     :param messages: list of messages from a channel
     :param topk: amount of messages to return from each channel
+    :param criterion: criterion for message sorting (replies / length / reactions)
     :return: sorted messages
     """
 
-    return sorted(messages, key=lambda x: x.get("reply_count", 0), reverse=True)[:topk]
+    if criterion not in {"replies", "length", "reactions"}:
+        logging.warning(f"Wrong criterion: {criterion}, using 'replies'")
+        criterion = "replies"
+
+    if criterion == "replies":
+        key = "reply_count"
+    elif criterion == "length":
+        key = "char_length"
+    else:
+        logging.info("Ha-ha, reactions are not supported yet, using replies")
+        key = "reply_count"
+
+    return sorted(messages, key=lambda x: x.get(key=key, default=0), reverse=True)[
+        :topk
+    ]
+
+
+async def process_message(message: dict) -> None:
+    """Answer only on needed messages"""
+
+    # do not answer on own messages
+    if message["user"] == "digest-bot":
```

Comment:
Are you sure the name of bot everywhere will be a "digest-bot"?

Reply:
resolved